test/shared/src/main/scala/zio/test/TimeVariants.scala

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2019 John A. De Goes and the ZIO Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package zio.test
+
+import java.time.{ Instant, LocalDateTime, OffsetDateTime, ZoneOffset }
+
+import zio.duration.Duration
+import zio.random.Random
+
+trait TimeVariants {
+
+  /**
+   * A generator of ZIO duration values. Shrinks toward Duration.Zero.
+   */
+  final def anyFiniteDuration: Gen[Random, Duration] = Gen.long(0L, Long.MaxValue).map(Duration.Finite(_))
+
+  /**
+   * A generator java.time.Instant values.
+   */
+  final def anyInstant: Gen[Random, Instant] = instant(Instant.MIN, Instant.MAX)
+
+  /**
+   * A generator java.time.LocalDateTime values.
+   */
+  final def anyLocalDateTime: Gen[Random, LocalDateTime] = localDateTime(LocalDateTime.MIN, LocalDateTime.MAX)
+
+  /**
+   * A generator java.time.OffsetDateTime values.
+   */
+  final def anyOffsetDateTime: Gen[Random, OffsetDateTime] = offsetDateTime(OffsetDateTime.MIN, OffsetDateTime.MAX)
+
+  /**
+   * A generator of ZIO duration values inside the specified range: [min, max].
+   */
+  final def duration(min: Duration, max: Duration): Gen[Random, Duration] =
```

Comment:
`finiteDuration`? Should we add a generator of infinite values (it would just be a constant generator) and possibly a generator of duration values that produces both? Not sure what ratio of finite to infinite values would make sense.